doc/connectivity/networking/api/coap_client.rst

```diff
@@ -79,6 +79,33 @@ The following is an example of a very simple response handling function:
         }
     }
 
+CoAP options may also be added to the request by the application. The following is an example of
+the application adding a Block2 option to the initial request, to suggest a maximum block size to
+the server for a resource that it expects to be large enough to require a blockwise transfer (see
+RFC7959 Figure 3: Block-Wise GET with Early Negotiation).
+
+.. code-block:: c
+
+    static struct coap_client;
+    struct coap_client_request req = { 0 };
+    struct coap_client_option block2_option;
```

Comment:
Just a thought, instead of providing an initializer function for such a static structure, can we just provide a macro for it?

```suggestion

// In coap_client.h
#define COAP_OPTION_BLOCK2_INIT (struct coap_client_option){.code = COAP_OPTION_BLOCK2, .len = 1, .value[0] = coap_bytes_to_block_size(CONFIG_COAP_CLIENT_BLOCK_SIZE)}

// In app code
    struct coap_client_option block2_option = COAP_OPTION_BLOCK2_INIT;
```